packages/interfaces/src/receiver.ts

```diff
@@ -0,0 +1,17 @@
+import type { IDecodedMessage, IDecoder } from "./message.js";
+import type { Callback, ProtocolOptions } from "./protocols.js";
+
+type Unsubscribe<T> = () => T;
+type PubSubTopic = string;
+type ContentTopic = string;
+
+export type ActiveSubscriptions = Map<PubSubTopic, ContentTopic[]>;
+
+export interface IReceiver {
+  subscribe: <T extends IDecodedMessage>(
+    decoders: IDecoder<T>[],
+    callback: Callback<T>,
+    opts?: ProtocolOptions
+  ) => Unsubscribe<void> | Promise<Unsubscribe<Promise<void>>>;
```

Comment:
This is another thing I want to figure out - either make it sync or async for the interface.

Reply:
Well, it is expected to have both:
- Relay: all actions are local as the node relays all messages of the pubsub topic so listening (subscribe) or stopping to listen (unsubscribe) to given content topics are local actions hence it's sync
- Filter: filtering of subscription is done by remote node, hence it's async

I don't see the dichotomy as an issue.

Reply:
The problem here is that types are not reliable - we don't know if `subscribe` should be awaited or not, same goes for `Unsubscribe`.